Dockerfile

```diff
@@ -11,43 +11,37 @@ USER 0
 COPY package.json yarn.lock lerna.json tsconfig.json .prettierrc ./
 COPY apps ./apps
 COPY libs ./libs
-RUN chmod 0500 -R apps libs
 RUN apk add python3 make g++
 RUN sed -i s^https://registry.yarnpkg.com^$YARNREPO^g yarn.lock
 RUN yarn --frozen-lockfile --production --network-timeout 600000
 
 RUN yarn run build
-RUN chown -R node apps libs package.json
-RUN chmod 0400 package.json libs/interfaces/package.json tsconfig.json .prettierrc
-RUN chmod 0500 -R apps libs
 
 ### Production image
 
 FROM $BASE_CONTAINER as app
```

Comment:
I've read that article for my previous project- I hadn't consulted it since it was for python, not node, but that's a fair point. 
As far as size, the multi-stage build should (or possibly could, if it isn't) offset the bloat, but time spent building dependencies is definitely a valid concern.

To definitively answer what the best choice is, though, we have to find what dependencies are being built that could be pulled if we used another image, find how much space they account for, and how much space/time we'd save by switching to that other image. Additional research could be done for this, but I'm beginning to run short on time on this project myself.